app/controllers/twitter_users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class TwitterUsersController < ApplicationController
+  before_action :current_user
+
+  def subscribe
+    @screen_name = params[:screen_name]
+    if twitter_user_protected?
+      flash[:error] = "That user is protected; you cannot subscribe to their tweets."
+    else
+      if !twitter_user_exists?
+        # if twitter_user doesn't exist, create twitter_user and their tweets
+        create_twitter_user(@screen_name)
+        create_tweets(@twitter_user)
+      end
+      @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+      if @current_user.twitter_users.include?(@twitter_user)
+        #if current_user is already subscribed to this twitter_user, show error
+        flash[:error] = "You are already subscribed to this user."
+      else
+        # subscribe current_user to twitter_user
+        @current_user.twitter_users << @twitter_user
+      end
+    end
+    redirect_to :root
+  end
+
+  def unsubscribe
+    @screen_name = params[:screen_name]
+    @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+    @current_user.twitter_users.delete(@twitter_user)
+    redirect_to :subscriptions
+  end
+
+  private
+
+  def twitter_user_exists?
+    if TwitterUser.find_by(screen_name: @screen_name)
+      return true
+    else
+      return false
+    end
+  end
+
+  def twitter_user_protected?
+    $client.user(@screen_name).protected?
+  end
+
+  def create_twitter_user(screen_name)
```

Comment:
I highly suggest you have a client class and a helper class that handle ALL interaction between your app and the API you're using. This type of functionally doesn't typically belong in the controller. My suggestion for flow would be: 

**Controller** (asking for data) -> **API Client** (makes call to API and calls API Helper to reformat data) -> **API Helper** (re-formats the data in the way you need to use it in your app and returns it back to API Client and then the API Client back to the Controller)

The controller can call a method from your client class, but the client would be solely responsible for 1. calling the api for data and 2. calling